src/sentry/eventstream/kafka/synchronized.py

```diff
@@ -0,0 +1,316 @@
+import logging
+from contextlib import contextmanager
+from datetime import datetime
+from threading import Event, Lock
+from time import time
+from typing import (
+    Callable,
+    Generic,
+    Iterator,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Sequence,
+    Set,
+    TypeVar,
+)
+
+from arroyo.backends.abstract import Consumer
+from arroyo.backends.kafka import KafkaPayload
+from arroyo.commit import CommitCodec
+from arroyo.errors import ConsumerError, EndOfPartition
+from arroyo.types import Message, Partition, Position, Topic, TPayload
+from arroyo.utils.concurrent import execute
+from arroyo.utils.metrics import get_metrics
+
+logger = logging.getLogger(__name__)
+
+DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+
+commit_codec = CommitCodec()
+
+
+T = TypeVar("T")
+
+
+class Synchronized(Generic[T]):
+    """
+    This class wraps a value that is shared between multiple threads,
+    providing thread-safe ``get`` and ``set`` methods for reading and writing
+    (replacing) the value.
+    """
+
+    def __init__(self, value: T, lock: Optional[Lock] = None) -> None:
+        if lock is None:
+            lock = Lock()
+
+        self.__value = value
+        self.__lock = lock
+
+    # TODO: For future use, it might make sense to expose the other lock
+    # arguments on `get` and `set`, such as `timeout`, `block`, etc.
+
+    @contextmanager
+    def get(self) -> Iterator[T]:
```

Comment:
this should be `Generator[T, None, None]` -- technically `contextmanager` can't take a raw Iterator since it needs  a  `.throw` attribute.  annoyingly the typeshed devs are against fixing this despite it being a common source of bugs

Reply:
Thanks!